src/picongpu/include/plugins/output/images/PngCreator.tpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2013-2016 Axel Huebl, Heiko Burau, Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+#include "simulation_defines.hpp"
+#include "plugins/output/images/PngCreator.hpp"
+#include "memory/boxes/DataBox.hpp"
+#include "plugins/output/header/MessageHeader.hpp"
+#include "verify.hpp"
+
+#include <string>
+#include <iostream>
+#include <sstream>
+#include <iomanip>
+
+#if( PIC_ENABLE_PNG == 1 )
+#   include <pngwriter.h>
+#endif
+
+namespace picongpu
+{
+    template< class Box >
+    inline void PngCreator::createImage(
+        const Box data,
```

Comment:
You could try to add `BOOST_GPU_ENABLED` to [`boost::ignore_unused`](https://github.com/boostorg/core/blob/develop/include/boost/core/ignore_unused.hpp) like:
```
template <typename... Ts>
BOOST_FORCEINLINE BOOST_GPU_ENABLED BOOST_CXX14_CONSTEXPR void ignore_unused(Ts const& ...)
{}
```
If this would work and could be integrated into boost, alpaka would be happy too 😄